packages/swap/src/routes/v2/swap.ts

```diff
@@ -0,0 +1,323 @@
+
+    const [
+      swapEgressFields,
+      refundEgressFields,
+      estimatedDurationSeconds,
+      srcChainRequiredBlockConfirmations,
+    ] = await Promise.all([
+      getEgressStatusFields(
+        swapEgress,
+        swapEgress?.broadcast,
+        ignoredEgresses,
+        'SWAP',
+        swapEgressTrackerTxRef,
+      ),
+      getEgressStatusFields(
+        refundEgress,
+        refundEgress?.broadcast,
+        ignoredEgresses,
+        'REFUND',
+        refundEgressTrackerTxRef,
+      ),
+      srcAsset && destAsset && estimateSwapDuration({ srcAsset, destAsset }),
+      internalSrcAsset && getRequiredBlockConfirmations(internalSrcAsset),
+    ]);
+
+    const response = {
+      state,
+      swapId: swapRequest?.nativeId.toString(),
+      ...(internalSrcAsset && getAssetAndChain(internalSrcAsset, 'src')),
+      ...(internalDestAsset && getAssetAndChain(internalDestAsset, 'dest')),
+      destAddress: readField(swapRequest, swapDepositChannel, failedSwap, 'destAddress'),
+      depositChannel: {
+        createdAt: swapDepositChannel?.createdAt.valueOf(),
+        brokerCommissionBps: swapDepositChannel?.brokerCommissionBps,
+        depositAddress: swapDepositChannel?.depositAddress,
+        expiryBlock: swapDepositChannel?.srcChainExpiryBlock?.toString(),
+        estimatedExpiryTime: swapDepositChannel?.estimatedExpiryAt?.valueOf(),
+        expectedDepositAmount: swapDepositChannel?.expectedDepositAmount?.toFixed(),
+        isExpired: swapDepositChannel?.isExpired,
+        openedThroughBackend: swapDepositChannel?.openedThroughBackend,
+        affiliateBrokers,
+        fillOrKillParams: swapDepositChannel?.fokMinPriceX128
+          ? {
+              retryDurationBlocks: swapDepositChannel.fokRetryDurationBlocks,
+              refundAddress: swapDepositChannel.fokRefundAddress,
+              minPrice: getPriceFromPriceX128(
+                swapDepositChannel.fokMinPriceX128.toFixed(),
+                swapDepositChannel.srcAsset,
+                swapDepositChannel.destAsset,
+              ),
+            }
+          : undefined,
+        dcaParams: swapDepositChannel?.chunkIntervalBlocks
+          ? {
+              numberOfChunks: swapDepositChannel?.numberOfChunks,
+              chunkIntervalBlocks: swapDepositChannel?.chunkIntervalBlocks,
+            }
+          : undefined,
+      },
+      deposit: {
+        amount:
+          readField(swapRequest, failedSwap, 'depositAmount')?.toFixed() ?? pendingDeposit?.amount,
+        transactionRef: depositTransactionRef,
+        txConfirmations: pendingDeposit?.transactionConfirmations,
+        receivedAt: swapRequest?.depositReceivedAt?.valueOf(),
+        receivedBlockIndex: swapRequest?.depositReceivedBlockIndex ?? undefined,
+        ...(failedSwap && { failure: getDepositIgnoredFailedState(failedSwap) }),
+      },
+      swap: {
+        ...rolledSwaps,
+        totalAmountSwapped: rolledSwaps?.totalAmountSwapped.toFixed(),
```

Comment:
its not clear if this is the srcAsset or the destAsset. maybe we should have a `inputAmountSwapped` and a `outputAmountSwapped`?

Reply:
also fine for me to call it `totalInputAmountSwapped`/`totalOutputAmountSwapped` so the prefix matches with `totalChunksExecuted`